src/urllib3/util/timeout.py

```diff
@@ -120,10 +121,12 @@ def __repr__(self) -> str:
     # __str__ provided for backwards compatibility
     __str__ = __repr__
 
+    @staticmethod
+    def resolve_default_timeout(timeout: _TYPE_TIMEOUT) -> _TYPE_TIMEOUT_RESOLVED:
+        return getdefaulttimeout() if timeout is _Default else timeout  # type: ignore[return-value]
```

Comment:
Thanks @bluetech  for your great suggestion :+1: . I  didn't know about this.

> _Default: "Fina[_TYPE_DEFAULT]l" = _TYPE_DEFAULT.token

I didn't understand it completely. what is the `Fina[_TYPE_DEFAULT]l` meaning? is it a typo?

Reply:
> I didn't understand it completely. what is the Fina[_TYPE_DEFAULT]l meaning? is it a typo?

Oops yes, I meant to write `Final` instead of `Fina`.

`Final` was added in Python 3.8. Since we support earlier than this, we need to import from `typing_extensions`. And since we don't have a runtime dep on `typing-extensions`, we should import like

```py
if TYPE_CHECKING:
    from typing_extensions import Final
```

and use it as `"Final"` instead of `Final`.

Reply:
Let's go ahead with using `Final` with a single-value `Enum` until we're provided another option. What a great way to handle this case! Thanks for the great suggestion @bluetech.